### DISCUSSION TITLE
v0.11.4

### DISCUSSION THREAD

**Discussion by gerwaric** [gerwaric/acquisition#70] [Announcements]
Now that I've hit 36/40 challenges for the Kalgurian portal, it's time for progresson acquisition.

### Bugfixes

- Fixed https://github.com/gerwaric/acquisition/issues/68, which was due to acquisition not properly handing tabs download via both the legacy API and the OAuth api.

### Changes

- As one user discovered in https://github.com/gerwaric/acquisition/issues/67, current offer buyouts (indicated by "c/o") are not indexed by the trade site. This kind of buyout appears to have falled out of use, because the most recent "c/o" buyout I can find is from years ago. Therefore, acquisition v0.11.4 will warn when it encounters a current offer buyout, but continue to run.

**WARNING:** support for "c/o" buyouts will be removed in the next major version. Let me know if you think they need to stay.

- Changed the Linux build to use the OpenSSL toolkit provided by QT, which must be installed via the Qt Maintenance Tool. This only simplifies the build setup. Users and developers both will still have to ensure compatible OpenSSL libraries are installed in a directory on the LD_LIBRARY_PATH.

## Roadmap

Here's where my head is at on what's next for acquisition. Open to feedback as well as contributions.

### Internal Rework

I'm revamping the way acquisition stores data, because the current method is designed around the legacy API. This design even predates the unique ids GGG now provides with every item, charcter, and stash tab. Aside from giving me headaches, it has several drawbacks:

- Acquisition does not handle buyouts on stash with the same name.
- It's currently difficult to keep track of item locations internally. For example, the database keys used to store buyouts are based on the internally-computed hash, so there's no way to manually debug the data stores, e.g. with a SQL viewer.
- Because acquisition computes it's own unique id for each item, those hashes depend on the exact formatting of the json received by the api, so even minor formatting changes (or changes made by acquisition, e.g. for compatibility) can cause buyouts to become "lost".

A redesign here will simplify the internals, a lot. However, it means changing the data storage format. This would mean that existing data files will no longer be useable. If this turns out to be the case, I'm hoping to add an import function to load data from older files.

### Possible Features

Here's the features I think acquisition should have someday:

- Support for searching for pseudo mods like "+% total to Chaos Resistance"
- Support for map tabs, unique tabs, and other tabs acquisition currently ignores.
- Dividing items into buckets for more effecient searching:
    * Equipable items such as weapons, armour, jewelry, flasks, tinctures, jewels, etc
    * Non-stackable items such as maps, blueprints, and inscribed ultimatums
    * Stackable items such as those tradeable in the 3.25 currency exchange

### Build Simplification

Right now the build process is clunky and fragile.

<hr /><em>This discussion was created from the release <a href='https://github.com/gerwaric/acquisition/releases/tag/v0.11.4'>v0.11.4</a>.</em>

**Comment by wallacrs** [comment]
Hmmm, it seems that the Windows installer for 11.4 is still installing the 11.3 executable file. That's what shows in the Details tab of the .exe file properties, and the Login prompt on startup shows Acquisition Login [0.11.3] as well. Hopefully a quick repack to fix...
BTW, a +1 from me on the pseudo mods, I had hacked the old v10 versions with the hardcoded modlist to include things like "total elemental resistance" which I found very handy when searching my stash tabs. Thanks again for all the work you're putting in to keep this great program going!

**Reply by gerwaric** [reply]
I probably made a mistake building the installer. I'll take a look in a couple hours. Thank you for the heads up.

**Reply by gerwaric** [reply]
@wallacrs Can you try again now?  I removed all the build directories, rebuilt from scratch, and checked the installer on my own computer to make sure it was installing 0.11.4. Please let me know if you're still seeing the problem.

PS - I also really want pseudo-mods to work. There was some code that suggested they might have worked in the past, but I'm not sure when it broke.

**Reply by wallacrs** [reply]
Yep, I have 11.4 now, and it seems to be working fine. Thanks!

**Reply by gerwaric** [reply]
Glad to hear it!

Ironically, another point release may be coming tomorrow, because I just heard back from GGG support about a few questions I had asked and forgotten about. They gave me some information I can use to parse certain legacy stash tabs (e.g. from 2015), as well as some info on rate limiting that I can use to avoid mysterious rate limit violations I was getting occaionally.
